src/main/java/com/gulteking/mqttbackendserver/repository/MqttTopicsRepository.java

```diff
@@ -3,9 +3,14 @@
 import com.gulteking.mqttbackendserver.entity.MqttTopics;
 import org.springframework.data.jpa.repository.JpaRepository;
 
+import java.util.List;
 import java.util.Optional;
 
 public interface MqttTopicsRepository extends JpaRepository<MqttTopics, Long> {
-    Optional<MqttTopics> findByMtName(String name);
-}
+    Optional<MqttTopics> findByMtSubscriberTopic(String name);
 
+    Optional<MqttTopics> findByMtPublisherTopic(String name);
+
+    List<MqttTopics> findByMtIsDeviceConnected(Boolean isConnected);
```

Comment:
I don't think we will ever pass false value to this method.
So you can remove the parameters and change the method to **findByMtIsDeviceConnectedTrue()**
This should work.

Reply:
I think this method is not used at all.

Reply:
This is for future use